### DISCUSSION TITLE
First timer and extremely stuck.  Have no idea what all this means.  Assistance would be very appreciated

### DISCUSSION THREAD

**Discussion by Bugseym** [imartinez/privateGPT#1556] [Q&A]
Hi Guys, I am someone with zero coding experience, however, recently it's something I would like to learn more about and AI seems like a cool place to start.  So I started with the PrivateGPT project and am now lost.  Hoping any one out there can help a novice out.  Thanks in advance. 

I followed all the installation steps and it all seemed to well using my 2017 iMac i5 5k.  Then once I entered the UI it just won't work.  Terminal shows the following, which means nothing to me at all.  
<img width="1011" alt="Screenshot 2024-01-30 at 01 52 27" src="https://github.com/imartinez/privateGPT/assets/157659734/ed786e71-5112-4e41-945a-9c5cc325b63a">

**Comment by ghost** [comment]
I am a newby myself. But feeding your image to GPT-4 lead to this response. I hope it helps:

    ggml_metal_graph_compute_block_invoke: error: unsupported op 'RMS_NORM': It seems that a specific operation (RMS_NORM) is not supported. This could be due to the version of a library you are using not supporting this operation on the hardware you are running it on, possibly because of Metal (Apple's graphics API).

    ggml_metal_graph_compute_block_invoke: error: unsupported op 'MUL_MAT': Similar to the first error, another operation (MUL_MAT) is also not supported.

    Abort trap: 6: This is a generic error message that usually indicates a crash in the program, possibly as a result of the unsupported operations mentioned above.

    resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown: This warning is from Python's multiprocessing library, which is informing you that there were resources (semaphores in this case) that weren't properly cleaned up before the program was shut down. This is a warning, not an error, and it indicates that some resources may not have been released back to the system.

The errors seem to be related to the specific operations that are not supported by the Metal API, or the libraries interfacing with it aren't handling them correctly. This might be due to incompatibilities between the library and the Metal API version, or it could be related to the way the Metal backend is implemented in the library you are using.

To resolve these errors, you would typically:

    Check if you are using the latest version of the libraries and update them if necessary.
    Review the documentation of the libraries you are using to ensure that the operations are supported and that you are using them correctly.
    If you are the developer of the library, you may need to handle these operations differently or provide an alternative implementation for them when running on Metal.

For a more precise solution, you would need to look into the specific libraries and the context in which these operations are being called.

**Comment by gshipley** [comment]
Are you running this in a VM by any chance?
